fe/fe-core/src/main/java/org/apache/doris/datasource/property/fileformat/CsvFileFormatProperties.java

```diff
@@ -0,0 +1,195 @@
+// specific language governing permissions and limitations
+// under the License.
+
+package org.apache.doris.datasource.property.fileformat;
+
+import org.apache.doris.analysis.Separator;
+import org.apache.doris.catalog.Column;
+import org.apache.doris.common.util.FileFormatUtils;
+import org.apache.doris.common.util.Util;
+import org.apache.doris.datasource.property.constants.CsvProperties;
+import org.apache.doris.nereids.exceptions.AnalysisException;
+import org.apache.doris.proto.InternalService.PFetchTableSchemaRequest;
+import org.apache.doris.qe.ConnectContext;
+import org.apache.doris.thrift.TFileAttributes;
+import org.apache.doris.thrift.TFileFormatType;
+import org.apache.doris.thrift.TFileTextScanRangeParams;
+import org.apache.doris.thrift.TResultFileSinkOptions;
+import org.apache.doris.thrift.TTextSerdeType;
+
+import com.google.common.base.Strings;
+import com.google.common.collect.Lists;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+import java.util.List;
+import java.util.Map;
+
+public class CsvFileFormatProperties extends FileFormatProperties {
+    public static final Logger LOG = LogManager.getLogger(
+            org.apache.doris.datasource.property.fileformat.CsvFileFormatProperties.class);
+
+    private String headerType = "";
+    private TTextSerdeType textSerdeType = TTextSerdeType.JSON_TEXT_SERDE;
+    private String columnSeparator = CsvProperties.DEFAULT_COLUMN_SEPARATOR;
+    private String lineDelimiter = CsvProperties.DEFAULT_LINE_DELIMITER;
+    private boolean trimDoubleQuotes;
+    private int skipLines;
+    private byte enclose;
+
+    // used by tvf
+    // User specified csv columns, it will override columns got from file
+    private final List<Column> csvSchema = Lists.newArrayList();
+
+    String defaultColumnSeparator = CsvProperties.DEFAULT_COLUMN_SEPARATOR;
+
+    public CsvFileFormatProperties(TFileFormatType fileFormatType) {
+        super(fileFormatType);
+    }
+
+    public CsvFileFormatProperties(TFileFormatType fileFormatType, String defaultColumnSeparator,
+            TTextSerdeType textSerdeType) {
+        super(fileFormatType);
+        this.defaultColumnSeparator = defaultColumnSeparator;
+        this.textSerdeType = textSerdeType;
+    }
+
+    public CsvFileFormatProperties(TFileFormatType fileFormatType, String headerType) {
+        super(fileFormatType);
+        this.headerType = headerType;
+    }
+
+
+    @Override
+    public void analyzeFileFormatProperties(Map<String, String> formatProperties, boolean isRemoveOriginProperty)
+            throws AnalysisException {
+        try {
+            // check properties specified by user -- formatProperties
+            columnSeparator = getOrDefaultAndRemove(formatProperties, CsvProperties.PROP_COLUMN_SEPARATOR,
+                    defaultColumnSeparator, isRemoveOriginProperty);
+            if (Strings.isNullOrEmpty(columnSeparator)) {
```

Comment:
It's unnecessary, because you already give a default value. So the column separator won't be empty.

Reply:
Same for other properties with default values